Source/CBLDatabase+Attachments.m

```diff
@@ -424,10 +432,16 @@ - (CBL_Revision*) updateAttachment: (NSString*)filename
     oldRev.properties = properties;
 
     // Store a new revision with the updated _attachments:
-    CBL_Revision* newRev = [self putRevision: oldRev prevRevisionID: oldRevID
-                             allowConflict: NO status: outStatus];
+    CBL_Revision* newRev = [self putRevision: oldRev prevRevisionID: oldRevID allowConflict: NO
+                                      status: outStatus error: outError];
     if (!body && *outStatus == kCBLStatusCreated)
         *outStatus = kCBLStatusOK;
+
+    if (CBLStatusIsError(*outStatus)) {
+        if (outError && !*outError)
```

Comment:
Ah, just noticed that `outError` is passed to the `putRevision:` call above. Hm. I think that means that `putRevision:` should have already put an NSError into it and this method shouldn't touch it.